### PR TITLE
Use better collections we possible

### PR DESCRIPTION
**Pull request by systay**
When we can use a primitive map, let's do that. This is also a
very common use of NodeHashJoin

### CODE REVIEW

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipe.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          copyDataFromRhs(newRow, currentRhsRow)
+          return Some(newRow)
+        }
+
+        while (rhsInput.nonEmpty) {
+          currentRhsRow = rhsInput.next()
+          val nodeId = currentRhsRow.getLongAt(lhsOffset)
+          if(nodeId != -1) {
+            val innerMatches = probeTable.get(nodeId)
+            if(innerMatches != null) {
```
Comment (fickludd) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipe.scala:100]: missing space

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipe.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          val newRow = SlottedExecutionContext(slots)
+          lhs.copyTo(newRow)
+          copyDataFromRhs(newRow, currentRhsRow)
+          return Some(newRow)
+        }
+
+        while (rhsInput.nonEmpty) {
+          currentRhsRow = rhsInput.next()
+          val nodeId = currentRhsRow.getLongAt(lhsOffset)
+          if(nodeId != -1) {
```
Comment (fickludd) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipe.scala:98]: missing space

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipe.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      }
+    }
+
+  private def copyDataFromRhs(newRow: SlottedExecutionContext, rhs: ExecutionContext): Unit = {
+    longsToCopy foreach {
+      case (from, to) => newRow.setLongAt(to, rhs.getLongAt(from))
+    }
+    refsToCopy foreach {
+      case (from, to) => newRow.setRefAt(to, rhs.getRefAt(from))
+    }
```
Comment (fickludd) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipe.scala:117]: As this is a very hot place, would doing the faster while-loop be worthwhile?

File: enterprise/cypher/slotted-runtime/src/test/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipeTest.scala [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+import scala.collection.immutable
+
+class NodeHashJoinSlottedPrimitivePipeTest extends CypherFunSuite {
+
+  test("should support simple hash join over nodes") {
+    // given
+    val node1 = 1
+    val node2 = 2
+    val node3 = 3
```
Comment (fickludd) [inline enterprise/cypher/slotted-runtime/src/test/scala/org/neo4j/cypher/internal/runtime/slotted/pipes/NodeHashJoinSlottedPrimitivePipeTest.scala:39]: These node are already private vals, so could be removed here.

**Review by fickludd — APPROVED** [review]
I have only nitpicks here, looks very good. I guess the side-effect of all this primitive-collection work you've been doing, is that you seem to have gotten a lot of knowledge of the new eclipse collections :).

Did you collect any benchmark data on this branch?

**Comment by systay** [comment]
moved over

**Pull request closed without merging**